### PR TITLE
[Messenger] Remove dependency on bridge packages

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? |no
| Tickets       | 
| License       | MIT
| Doc PR        | 

This removes code deprecated in 5.1. This will force the user to install the bridge instead of only the symfony/messenger

### CODE REVIEW

File: src/Symfony/Component/Messenger/composer.json [hunk 1]
```diff
@@ -18,10 +18,7 @@
     "require": {
         "php": ">=8.0.2",
         "psr/log": "~1.0",
-        "symfony/amqp-messenger": "^5.1|^6.0",
         "symfony/deprecation-contracts": "^2.1",
```
Comment (jderusse) [inline src/Symfony/Component/Messenger/composer.json]: can be removed too

**Comment by derrabus** [comment]
This change deserves a ChangeLog entry.

**Review by jderusse — COMMENTED** [review]
Don't forget the CHANGELOG ;)

File: src/Symfony/Component/Messenger/composer.json (hunk 1 above)
Comment (Nyholm) [reply src/Symfony/Component/Messenger/composer.json]: No. There are still deprecations. I'll work with them in a followup PR.

**Comment by Nyholm** [comment]
Thank you. 
I've added a changelog. 

The upgrade docs already included this change.

File: src/Symfony/Component/Messenger/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+6.0
+---
+
+ * Remove deprecated classes in `Symfony/Component/Messenger/Transport/AmqpExt`, `Symfony/Component/Messenger/Transport/Doctrine` and `Symfony/Component/Messenger/Transport/Redis`.
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/CHANGELOG.md]:
```suggestion
 * Remove deprecated classes `Symfony/Component/Messenger/Transport/AmqpExt`, `Symfony/Component/Messenger/Transport/Doctrine` and `Symfony/Component/Messenger/Transport/Redis`.
```

**Review by derrabus — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @Nyholm.

**Pull request merged**